WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -391,39 +395,68 @@ public async Task<ConnectionConfirmationResponse> ConfirmConnectionAsync(Connect
 					throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.IncorrectRequestedAmountCredentials, $"Round ({request.RoundId}): Incorrect requested amount credentials.");
 				}
 
-				var commitAmountZeroCredentialResponse = round.AmountCredentialIssuer.PrepareResponse(request.ZeroAmountCredentialRequests);
-				var commitVsizeZeroCredentialResponse = round.VsizeCredentialIssuer.PrepareResponse(request.ZeroVsizeCredentialRequests);
-
-				if (round.Phase == Phase.InputRegistration)
+				if (round.Phase is not (Phase.InputRegistration or Phase.ConnectionConfirmation))
```

Comment:
This validation is so cheap that I would move it up immediately after validating that the round really exists.

Reply:
ACK